src/freenet/node/NodeFile.java

```diff
@@ -0,0 +1,89 @@
+package freenet.node;
+
+import java.io.File;
+
+/**
+ * Mapping of files managed by the node to their respective locations.
+ */
+public enum NodeFile {
+    Seednodes(InstallDirectory.Node, "seednodes.fref"),
+    InstallerWindows(InstallDirectory.Run, "freenet-latest-installer-windows.exe"),
+	InstallerNonWindows(InstallDirectory.Run, "freenet-latest-installer-nonwindows.jar"),
+	IPv4ToCountry(InstallDirectory.Run, "IpToCountry.dat");
+
+    private final InstallDirectory dir;
+    private final String filename;
+    
+    /**
+     * Gets the absolute file path associated with this file for the given node instance.
+     */
+    public File getFile(Node node) {
+        return dir.getDir(node).file(filename);
+    }
+    
+    /**
+     * Gets the filename associated with this file.
+     */
+    public String getFilename() {
+        return filename;
+    }
+    
+    /**
+     * Gets the base directory with this file for the given node instance.
+     */
+    public ProgramDirectory getProgramDirectory(Node node) {
+        return dir.getDir(node);
+    }
+    
+    private NodeFile(InstallDirectory dir, String filename) {
+        this.dir = dir;
+        this.filename = filename;
+    }
+    
+    private enum InstallDirectory {
```

Comment:
Looks like it's redundant to mark an `enum` constructor private?

> The constructor for an enum type must be package-private or private access. It automatically creates the constants that are defined at the beginning of the enum body. You cannot invoke an enum constructor yourself. 

-- [Enum Types](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)